app/src/main/java/com/example/util/simpletimetracker/wear/WearService.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package com.example.util.simpletimetracker.wear
+
+import com.example.util.simpletimetracker.domain.interactor.PrefsInteractor
+import com.example.util.simpletimetracker.domain.interactor.RecordTagInteractor
+import com.example.util.simpletimetracker.domain.interactor.RecordTypeInteractor
+import com.example.util.simpletimetracker.domain.interactor.RunningRecordInteractor
+import com.example.util.simpletimetracker.domain.mapper.AppColorMapper
+import com.example.util.simpletimetracker.wearrpc.WearRPCServer
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.tasks.Tasks
+import com.google.android.gms.wearable.WearableListenerService
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.runBlocking
+import javax.inject.Inject
+
+/**
+ * Landing point for messages coming to Mobile from Wear.
+ *
+ * Implemented as a Service so that it can receive messages even when the Mobile app is stopped or
+ * in the background.
+ */
+@AndroidEntryPoint
+class WearService : WearableListenerService() {
+
+    @Inject
+    lateinit var prefsInteractor: PrefsInteractor
+
+    @Inject
+    lateinit var recordTypeInteractor: RecordTypeInteractor
+
+    @Inject
+    lateinit var recordTagInteractor: RecordTagInteractor
+
+    @Inject
+    lateinit var runningRecordInteractor: RunningRecordInteractor
+
+    @Inject
+    lateinit var appColorMapper: AppColorMapper
+
+    override fun onRequest(nodeId: String, path: String, request: ByteArray): Task<ByteArray>? {
+        val rpc = WearRPCServer(
```

Comment:
Probably true.

I chose this approach because I've been burned by confusing dependency injection frameworks, so I typically prefer to do manual dependency injection wherever possible. I find the plainness of the implementation more understandable than needing to reason about how the DI framework assembles a dependency graph and "magically" supplies things.

The other benefit of avoiding `@Inject`ions on `WearRPCServer` is that I can supply a different `SimpleTimeTrackerAPI` during [unit tests for testing key parts of the watch/phone communication](https://github.com/Razeeman/Android-SimpleTimeTracker/pull/162/files#diff-54c34b24ae115079cdf62a71b1a64bfd878fa6481ba2719ad7918e670bfb9f09) without running on an emulator.

All that said, as long as the code remains easily testable, I'm very open to adjustments to this implementation.

Reply:
Sure, I see your point. This could be changed later.